runtimes/common/src/migrations.rs

```diff
@@ -16,67 +16,77 @@
 
 // If you feel like getting in touch with us, you can do so at info@botlabs.org
 
-use sp_std::marker::PhantomData;
-
 use frame_support::{
-	traits::{Get, OnRuntimeUpgrade},
-	StorageHasher, Twox128,
+	pallet_prelude::ValueQuery,
+	storage_alias,
+	traits::{GetStorageVersion, OnRuntimeUpgrade},
 };
+use sp_runtime::traits::{Get, Zero};
+use sp_std::marker::PhantomData;
+
+use ctype::{CtypeCreatorOf, CtypeEntryOf};
+
+#[storage_alias]
+type MigrationCounter<T: ctype::Config> = StorageValue<ctype::Pallet<T>, u32, ValueQuery>;
```

Comment:
This is not the correct way of using temp storage in a migration.

In our current version of Polkadot, the pre-hook would use `Self::set_temp_storage(migration_counter, "ctype_migration_counter");` and post `Self::get_temp_storage("ctype_migration_counter")`.

After #434 in Polkadot v0.9.32+ the pre-hook actually returns any temp storage and the post one takes this as argument.

Reply:
I took this from the [Parachain staking migration](https://github.com/KILTprotocol/kilt-node/blob/621997ed4bed76a30aa9a53fdc21ef0217e3496e/pallets/parachain-staking/src/migration.rs#L65), and I thought it was a nice approach. Plus, this PR was worked on before the 0.9.32 upgrade was in, which is now, so I will probably use the new approach.

Reply:
How ironic the inspiration was taken by me, the one who complains 😅 Maybe the temp storage pre `0.9.32` was deprecated then. In any case, since it is only used in the `try-runtime` hooks, it would be completely fine to leave as is. Sorry about the noise.

Reply:
https://github.com/KILTprotocol/kilt-node/pull/440/commits/4d92056e7b37a14655f461da57441795c1a89605